xdsl/dialects/csl/csl.py

```diff
@@ -400,7 +400,7 @@ def get_element_type(self):
         return self.res.type.get_element_type()
 
     @staticmethod
-    def from_type(child_type: TypeAttribute) -> VariableOp:
```

Comment:
Should this not be a `TypeAttribute`?

Reply:
Then pyright complains in this line: `[csl.VariableOp.from_type(arg_t) for arg_t in op.iter_args.types]`

Reply:
This seems like the wrong place to fix this then..?

Although, @AntonLydike, is there a reason some types do not inherit `TypeAttribute`? In which case this should be an `Attribute`.

Reply:
Well, good question. All types *should* inherit from `TypeAttribute`, as that will have them be printed as `!dialect.name` instead of `#dialect.name` (which is attribute specific and can't be used as a type according to MLIR).